### DISCUSSION TITLE
file name extraction

### DISCUSSION THREAD

**Discussion by sepidhk** [Project-MONAI/MONAI#7748] [General]
hi all,
in the previous version of MONAI, I use the
`fname= os.path.basename(test_data["img_meta_dict"]["filename_or_obj"][0])`
 to extract the file name from data dictionary and it seems like this key is not available now. I would appreciate it if you could advice how to get the file name.

**Comment by KumoLiu** [comment]
Hi @sepidhk, use `test_data.meta["filename_or_obj"]` instead.

We introduced `MetaTensor` since 0.9.0: https://github.com/Project-MONAI/MONAI/wiki/MetaTensor-guide

Thanks.

**Reply by sepidhk** [reply]
get the following error

`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[71], line 23
     20 predicted = torch.argmax(test_data["pred"], dim=1, keepdim=True).detach().cpu()
     21 predicted_mask= np.moveaxis(np.squeeze(predicted, 0), 0, -1)
---> 23 filename = test_data.meta["filename_or_obj"]
     25 predicted_mask[predicted_mask == 0] = 0
     26 predicted_mask[predicted_mask == 1] = 53

AttributeError: 'dict' object has no attribute 'meta'`

**Reply by sepidhk** [reply]
here is my data transform, reading the images using PILReader:
test_transforms_encoded = Compose([
    LoadImaged(keys=["img", "label"], reader=PILReader),
    EnsureChannelFirstd(keys=["img", "label"]),
    ScaleIntensityd(keys=["img"], allow_missing_keys=True),
    Flipd(keys=["img", "label"], spatial_axis=0),
    Rotate90d(keys=["img", "label"], k=3),    
    EnsureTyped(keys=["img", "label"]),
    AsDiscreted(keys=["label"], to_onehot=num_classes),    
])

test_ds_encoded = monai.data.Dataset(data=val_files, transform=test_transforms_encoded)
test_loader_encoded = DataLoader(test_ds_encoded, batch_size=1, shuffle=False, num_workers=1)

**Reply by sepidhk** [reply]
Solved by: 
filename = test_inputs.meta["filename_or_obj"]
